### PR TITLE
feat(github): add artifacthub linting

### PR DESCRIPTION
**Pull request by jmeridth**
This will help us ensure we use correct notation in our artifacthub annotations

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -45,6 +45,18 @@ jobs:
       - name: Run chart-testing (lint)
         run: ct lint --debug --config ./.github/configs/ct-lint.yaml --lint-conf ./.github/configs/lintconf.yaml
 
+      - name: Artifacthub lint
+        id: artifacthub-lint
+        if: steps.list-changed.outputs.changed == 'true'
+        run: |
+          echo "NOTE: The supported change kinds are added, changed, deprecated, removed, fixed and security" \
```
Comment (jmeridth) [inline .github/workflows/lint-and-test.yml:52]: Thought this would be helpful no matter pass or fail.  Especially if fail.

**Comment by jmeridth** [comment]
@mkilchhofer found [this](https://github.com/artifacthub/hub/blob/ac49ca921ac7c7711b03d0701f52c33acaaaa6f9/.github/workflows/ci.yml#L28-L37) which is so much cleaner.  Will switch.

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -6,6 +6,17 @@ permissions:
   contents: read
 
 jobs:
+  linter-artifacthub:
+    runs-on: ubuntu-20.04
+    container:
+      image: artifacthub/ah
```
[the remark this answers is not in the document]
Comment (jmeridth) [reply .github/workflows/lint-and-test.yml]: Decided to pin to tag

**Comment by jmeridth** [comment]
Nice, this worked as expected [here](https://github.com/argoproj/argo-helm/actions/runs/4919608987/jobs/8787444190#step:4:52) since #2032 hasn't merged yet.  Nice!

File: .github/workflows/lint-and-test.yml [hunk 3]
```diff
@@ -6,6 +6,18 @@ permissions:
   contents: read
 
 jobs:
+  linter-artifacthub:
+    runs-on: ubuntu-latest
+    container:
+      image: artifacthub/ah:v1.14.0
+      options: --user 1001
```
Comment (jmeridth) [inline .github/workflows/lint-and-test.yml:13]: requires non-root user

**Review by mkilchhofer — APPROVED** [review]
🚀

**Comment by jmeridth** [comment]
Now that #2032 has merged, this should pass.

**Comment by jmeridth** [comment]
@yu-croco could you please take a look at this? Would like in place so we don't do bad artifacthub annotations again.

**Review by yu-croco — APPROVED** [review]
Thank you!

**Pull request merged**